Source/Evolve/WarpXEvolve.cpp

```diff
@@ -355,15 +359,18 @@ WarpX::OneStep_nosub (Real cur_time)
 #else
         EvolveF(0.5*dt[0], DtType::FirstHalf);
         FillBoundaryF(guard_cells.ng_FieldSolverF);
-#ifdef WARPX_MAG_LLG
-        EvolveM(0.5*dt[0]); // we now have M^{n+1/2}
-#endif
+
         EvolveB(0.5*dt[0]); // We now have B^{n+1/2}
+        FillBoundaryB(guard_cells.ng_FieldSolver, IntVect::TheZeroVector());
 
```

Comment:
For the coupled M + E + B evolution, we are eventually :) going to want EvolveB to occur before EvolveM (since the 2nd-order evolveM relies on B at the 'old' and 'new' times).  You should be able to switch it now since we are doing M-only tests.

Reply:
That's right! 

EvolveM now happens after EvolveB. You can see two EvolveB(0.5*dt[0]) occurs right before two MacroscopicEvolveM(0.5*dt[0]) now. 

But yes, the notion of storing the previous B will be needed and let's add it after this PR.